### ISSUE TITLE
Error: IPMI call failed: power status on CentOS Linux release 7.7.1908 (Core)

### ISSUE THREAD

**Issue by Lukasz-SendGrid** [metal3-io/metal3-dev-env#161]
Hi, I'm currently trying to set up my dev env and keep hitting the error above. A bit new to the bare metal stuff so any help would be very appreciated.

This is what my virsh looks like:
```
sudo virsh list --all
 Id    Name                           State
----------------------------------------------------
 25    minikube                       running
 28    node_0                         running
 29    node_1                         running
```

Here's what `kubectl get bmh -n metal3` returns
```
NAME     STATUS   PROVISIONING STATUS   CONSUMER   BMC                         HARDWARE PROFILE   ONLINE   ERROR
node-0   error    registration error               ipmi://192.168.111.1:6230                      true     Failed to get power state for node be2287c0-0834-486d-b43c-5f1fe4aef4f7. Error: IPMI call failed: power status.
node-1   error    registration error               ipmi://192.168.111.1:6231                      true     Failed to get power state for node 55201488-7ee7-4e5f-97e8-c2b45897a76a. Error: IPMI call failed: power status.
```
Logs for the operator show the same story:
```
{"level":"info","ts":1576875459.2814622,"logger":"baremetalhost_ironic","msg":"validating management access","host":"node-1"}
{"level":"info","ts":1576875459.3040826,"logger":"baremetalhost_ironic","msg":"found existing node by ID","host":"node-1"}
{"level":"info","ts":1576875459.304359,"logger":"baremetalhost_ironic","msg":"current provision state","host":"node-1","lastError":"","current":"verifying","target":"manageable"}
{"level":"info","ts":1576875459.3044887,"logger":"baremetalhost","msg":"response from validate","Request.Namespace":"metal3","Request.Name":"node-1","provisioningState":"registering","provResult":{"Dirty":true,"RequeueAfter":0,"ErrorMessage":""}}
{"level":"info","ts":1576875459.3046327,"logger":"baremetalhost","msg":"host not ready","Request.Namespace":"metal3","Request.Name":"node-1","provisioningState":"registering","wait":0}
{"level":"info","ts":1576875459.304894,"logger":"baremetalhost","msg":"saving host status","Request.Namespace":"metal3","Request.Name":"node-1","provisioningState":"registering","operational status":"OK","provisioning state":"registering"}
{"level":"info","ts":1576875459.3119812,"logger":"baremetalhost","msg":"done","Request.Namespace":"metal3","Request.Name":"node-1","provisioningState":"registering","requeue":true,"after":0}
{"level":"info","ts":1576875459.312225,"logger":"baremetalhost","msg":"Reconciling BareMetalHost","Request.Namespace":"metal3","Request.Name":"node-1"}
{"level":"info","ts":1576875459.312614,"logger":"baremetalhost","msg":"registering and validating access to management controller","Request.Namespace":"metal3","Request.Name":"node-1","provisioningState":"registering","credentials":{"credentials":{"name":"node-1-bmc-secret","namespace":"metal3"},"credentialsVersion":"6943"}}
{"level":"info","ts":1576875459.3128545,"logger":"baremetalhost_ironic","msg":"validating management access","host":"node-1"}
{"level":"info","ts":1576875459.3331463,"logger":"baremetalhost_ironic","msg":"found existing node by ID","host":"node-1"}
{"level":"info","ts":1576875459.3331733,"logger":"baremetalhost_ironic","msg":"current provision state","host":"node-1","lastError":"","current":"verifying","target":"manageable"}
{"level":"info","ts":1576875459.3331816,"logger":"baremetalhost","msg":"response from validate","Request.Namespace":"metal3","Request.Name":"node-1","provisioningState":"registering","provResult":{"Dirty":true,"RequeueAfter":0,"ErrorMessage":""}}
{"level":"info","ts":1576875459.3331926,"logger":"baremetalhost","msg":"host not ready","Request.Namespace":"metal3","Request.Name":"node-1","provisioningState":"registering","wait":0}
{"level":"info","ts":1576875459.333346,"logger":"baremetalhost","msg":"saving host status","Request.Namespace":"metal3","Request.Name":"node-1","provisioningState":"registering","operational status":"OK","provisioning state":"registering"}
{"level":"info","ts":1576875459.340921,"logger":"baremetalhost","msg":"done","Request.Namespace":"metal3","Request.Name":"node-1","provisioningState":"registering","requeue":true,"after":0}
{"level":"info","ts":1576875459.9524033,"logger":"baremetalhost","msg":"Reconciling BareMetalHost","Request.Namespace":"metal3","Request.Name":"node-1"}
{"level":"info","ts":1576875459.9526906,"logger":"baremetalhost","msg":"registering and validating access to management controller","Request.Namespace":"metal3","Request.Name":"node-1","provisioningState":"registering","credentials":{"credentials":{"name":"node-1-bmc-secret","namespace":"metal3"},"credentialsVersion":"6943"}}
{"level":"info","ts":1576875459.9529614,"logger":"baremetalhost_ironic","msg":"validating management access","host":"node-1"}
{"level":"info","ts":1576875459.9883375,"logger":"baremetalhost_ironic","msg":"found existing node by ID","host":"node-1"}
{"level":"info","ts":1576875459.988423,"logger":"baremetalhost_ironic","msg":"current provision state","host":"node-1","lastError":"Failed to get power state for node 55201488-7ee7-4e5f-97e8-c2b45897a76a. Error: IPMI call failed: power status.","current":"enroll","target":""}
{"level":"info","ts":1576875459.9885383,"logger":"baremetalhost","msg":"response from validate","Request.Namespace":"metal3","Request.Name":"node-1","provisioningState":"registering","provResult":{"Dirty":false,"RequeueAfter":0,"ErrorMessage":"Failed to get power state for node 55201488-7ee7-4e5f-97e8-c2b45897a76a. Error: IPMI call failed: power status."}}
{"level":"info","ts":1576875459.9886603,"logger":"baremetalhost","msg":"saving host status","Request.Namespace":"metal3","Request.Name":"node-1","provisioningState":"registering","operational status":"error","provisioning state":"registration error"}
{"level":"info","ts":1576875459.9938457,"logger":"baremetalhost","msg":"publishing event","reason":"RegistrationError","message":"Failed to get power state for node 55201488-7ee7-4e5f-97e8-c2b45897a76a. Error: IPMI call failed: power status."}
{"level":"info","ts":1576875460.0001786,"logger":"baremetalhost","msg":"stopping on host error","Request.Namespace":"metal3","Request.Name":"node-1","provisioningState":"registering","message":"Failed to get power state for node 55201488-7ee7-4e5f-97e8-c2b45897a76a. Error: IPMI call failed: power status."}
{"level":"info","ts":1576875460.00066,"logger":"baremetalhost","msg":"Reconciling BareMetalHost","Request.Namespace":"metal3","Request.Name":"node-1"}
{"level":"info","ts":1576875460.0009577,"logger":"baremetalhost","msg":"stopping on host error","Request.Namespace":"metal3","Request.Name":"node-1","provisioningState":"registration error","message":"Failed to get power state for node 55201488-7ee7-4e5f-97e8-c2b45897a76a. Error: IPMI call failed: power status."}
```

`sudo vbmc list` returned:

```
+-------------+---------+---------+------+
| Domain name | Status  | Address | Port |
+-------------+---------+---------+------+
| node_0      | running | ::      | 6230 |
| node_1      | running | ::      | 6231 |
+-------------+---------+---------+------+
```
I've tried deleting the bmh crds and applying them again with the same result. Thoughts?

**Comment by vincent-pli** [comment]
@Lukasz-SendGrid 
Could you run the `sudo vbmc list` and append result.

**Comment by Lukasz-SendGrid** [comment]
hey @vincent-pli this is what `sudo vbmc list` returned:

```
+-------------+---------+---------+------+
| Domain name | Status  | Address | Port |
+-------------+---------+---------+------+
| node_0      | running | ::      | 6230 |
| node_1      | running | ::      | 6231 |
+-------------+---------+---------+------+
```

**Comment by vincent-pli** [comment]
@Lukasz-SendGrid 
I hit the same issue, you could take a try to modify the configuration of `vbmc` to change the address to `192.168.111.1` rather than `::`, then restart the `vbmc`.

It resolve my issue, good luck for you.

**Comment by ian-howell** [comment]
@Lukasz-SendGrid Did you ever resolve this issue?

I'm seeing this issue on an Ubuntu 18 installation as well. I changed https://github.com/metal3-io/metal3-dev-env/blob/master/vm-setup/roles/virtbmc/tasks/setup_tasks.yml#L78 from `::` to `192.168.111.1`, then ran `make clean; make`. I'm seeing the correct output from `sudo vbmc list`, but I'm still getting the "Failed to get power state for node" error.

**Comment by ian-howell** [comment]
This is probably fixed with #210.

Here's the quick and dirty fix:

```
sudo apt install openssh-server
```

**Comment by maelk** [comment]
Might be fixed in #233

**Comment by hardys** [comment]
I have also seen issues where permissions problems prevented vbmc to control the nodes (normally due to running the scripts as root, then switching to a regular user).

A helpful debug step beyond just checking the `vbmc list` is to check you can query the power status directly via ipmitool (first on the test host, then you can test the same from inside the ironic-conductor container on the cluster):

```
ipmitool -I lanplus -U admin -P password -H 192.168.111.1 -p 6230 power status
```

If that fails it's then easier to trigger the problem and debug e.g via the VBMC logs.
